runtime_scan/pkg/orchestrator/common/queue.go

```diff
@@ -108,9 +117,49 @@ func (q *Queue[T]) Enqueue(item T) {
 	q.l.Lock()
 	defer q.l.Unlock()
 
+	q.enqueue(item)
+}
+
+// EnqueueAfter will tell the queue to keep track of the item preventing it
+// being enqueued before the specified duration through Enqueue. It will then
+// Enqueue it after a defined duration. If item already in the queue, is
+// processing or waiting to be enqueued, nothing will be done.
+func (q *Queue[T]) EnqueueAfter(item T, d time.Duration) {
+	q.l.Lock()
+	defer q.l.Unlock()
+
+	q.enqueueAfter(item, d)
+}
+
+// Internal enqueueAfter function that it can be reused by the public
+// EnqueueAfter and public RequeueAfter functions, these should not be called
+// without obtaining a lock on Queue first.
+func (q *Queue[T]) enqueueAfter(item T, d time.Duration) {
+	_, inQueue := q.inqueue[item]
+	_, isProcessing := q.processing[item]
+	_, isWaitingForEnqueue := q.waitingForEnqueue[item]
+	if inQueue || isProcessing || isWaitingForEnqueue {
+		// item is already known by the queue so there is nothing to do
```

Comment:
@sambetts-cisco maybe a warning is needed here, it is probably a "bug" that you try to `enqueueAfter` for an already known item.